torchgeo/datasets/pastis.py

```diff
@@ -0,0 +1,444 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""PASTIS dataset."""
+
+import abc
+import os
+from collections.abc import Sequence
+from typing import Callable, Optional, Union
+
+import fiona
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from matplotlib.colors import ListedColormap
+from torch import Tensor
+
+from .geo import NonGeoDataset
+from .utils import check_integrity, download_url, extract_archive
+
+
+class PASTIS(NonGeoDataset, abc.ABC):
+    """PASTIS dataset.
+
+    The `PASTIS <https://github.com/VSainteuf/pastis-benchmark>`__
+    dataset is a dataset for time-series panoptic segmentation of agricultural parcels
+
+    Dataset features:
+
+    * support for the original PASTIS and PASTIS-R versions of the dataset
+    * 2,433 time-series with 10 m per pixel resolution (128x128 px)
+    * 18 crop categories, 1 background category, 1 void category
+    * semantic and instance annotations
+    * 3 Sentinel-1 Ascending bands
+    * 3 Sentinel-1 Descending bands
+    * 10 Sentinel-2 L2A multispectral bands
+
+    Dataset format:
+
+    * time-series and annotations are in numpy format (.npy)
+
+    Dataset classes:
+
+    0. Background
+    1. Meadow
+    2. Soft Winter Wheat
+    3. Corn
+    4. Winter Barley
+    5. Winter Rapeseed
+    6. Spring Barley
+    7. Sunflower
+    8. Grapevine
+    9. Beet
+    10. Winter Triticale
+    11. Winter Durum Wheat
+    12. Fruits Vegetables Flowers
+    13. Potatoes
+    14. Leguminous Fodder
+    15. Soybeans
+    16. Orchard
+    17. Mixed Cereal
+    18. Sorghum
+    19. Void Label
+
+    If you use this dataset in your research, please cite the following papers:
+
+    * https://doi.org/10.1109/ICCV48922.2021.00483
+    * https://doi.org/10.1016/j.isprsjprs.2022.03.012
+
+    .. versionadded:: 0.5
+    """
+
+    classes = [
+        "background",  # all non-agricultural land
+        "meadow",
+        "soft_winter_wheat",
+        "corn",
+        "winter_barley",
+        "winter_rapeseed",
+        "spring_barley",
+        "sunflower",
+        "grapevine",
+        "beet",
+        "winter_triticale",
+        "winter_durum_wheat",
+        "fruits_vegetables_flowers",
+        "potatoes",
+        "leguminous_fodder",
+        "soybeans",
+        "orchard",
+        "mixed_cereal",
+        "sorghum",
+        "void_label",  # for parcels mostly outside their patch
+    ]
+    cmap = ListedColormap(
```

Comment:
I know this isn't a GeoDataset, but I wonder if we want to follow the style of GeoDataset where `cmap` is a dict.

Reply:
I guess we can -- I just don't know of a situation where you want the colors _not_ in a ListedColormap